### PR TITLE
wip: update to storybook 7

### PR DESCRIPTION
**Pull request by bnjm**
This PR bumps several dependencies, the most significant change being the Storybook update to v7 (which uses Vite under the hood). The change was motivated by conflicting peer dependencies in #64, which this PR encompasses.

The main commit to review is [194fc49](https://github.com/archilogic-com/honeycomb/pull/66/commits/194fc4930082443a5c4d5049199cec09104b2f15). The automigrate helper (https://storybook.js.org/docs/react/migration-guide) got us 90% of the way there.

Included changes from the combine-dependabot PR:

https://github.com/archilogic-com/honeycomb/pull/61 chore: bump vite-plugin-dts from 3.0.3 to 3.5.1
https://github.com/archilogic-com/honeycomb/pull/60 chore: bump @vueuse/core from 10.2.1 to 10.3.0
https://github.com/archilogic-com/honeycomb/pull/59 chore: bump @types/lodash from 4.14.195 to 4.14.196
https://github.com/archilogic-com/honeycomb/pull/58 chore: bump @vitest/ui from 0.33.0 to 0.34.1
https://github.com/archilogic-com/honeycomb/pull/57 chore: bump @typescript-eslint/eslint-plugin from 5.59.11 to 6.3.0
https://github.com/archilogic-com/honeycomb/pull/56 chore: bump vitest from 0.32.2 to 0.34.1
https://github.com/archilogic-com/honeycomb/pull/54 chore: bump eslint from 8.45.0 to 8.46.0
https://github.com/archilogic-com/honeycomb/pull/53 chore: bump eslint-plugin-vue from 9.15.1 to 9.16.1
https://github.com/archilogic-com/honeycomb/pull/52 chore: bump prettier from 3.0.0 to 3.0.1

This closes AL-3484

### CODE REVIEW

File: src/stories/AppMenu.stories.mdx [hunk 1]
```diff
@@ -732,7 +732,6 @@ Menu items can be disabled.
         template: '<div class="absolute flex inset-0 p-16"><story /></div>'
       })
     ]}>
-    >
```
Comment (bnjm) [inline src/stories/AppMenu.stories.mdx:735]: dormant typo which threw an error due to the [mdx2 update](https://storybook.js.org/docs/vue/migration-guide#upgrade-mdx1-to-mdx2)

**Comment by romiekos** [comment]
Wow, new UI and the docs page
image

**Comment by romiekos** [comment]
And the syntax highlight!

image

**Comment by romiekos** [comment]
Some buttons are not working.
Example: go to https://649d652b7201581ed9f63642-pogfmcxpzj.chromatic.com/?path=/docs/components-dialog--docs and try to click on the "Show code" button.
image

File: .storybook/manager.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { addons } from '@storybook/manager-api'
+import { themes } from '@storybook/theming'
+
+addons.setConfig({
+  theme: themes.light
```
Comment (bnjm) [inline .storybook/manager.js:5]: without this, the new default for Storybook is 'auto', which looks weird next to our non-dark-mode-friendly design system. I also didn't want to change too much while Iana is away 😅

Screenshot 2023-08-09 at 14 23 36

**Comment by bnjm** [comment]
> Some buttons are not working. Example: go to [649d652b7201581ed9f63642-pogfmcxpzj.chromatic.com/?path=/docs/components-dialog--docs](https://649d652b7201581ed9f63642-pogfmcxpzj.chromatic.com/?path=/docs/components-dialog--docs) and try to click on the "Show code" button. image

good catch, that might be a blocker, will put this is back into draft and look into it

**Comment by ianaa** [comment]
> > Some buttons are not working. Example: go to [649d652b7201581ed9f63642-pogfmcxpzj.chromatic.com/?path=/docs/components-dialog--docs](https://649d652b7201581ed9f63642-pogfmcxpzj.chromatic.com/?path=/docs/components-dialog--docs) and try to click on the "Show code" button. image
> 
> good catch, that might be a blocker, will put this is back into draft and look into it

@romandkos @bnjm thanks for your time on this! After spending quite some time trying to resolve this issue I think the problem lies within the fact that storybook docs run on React, and us trying to render Vue templates inside of them results in an error. [here from the docs](https://storybook.js.org/docs/vue/writing-docs/mdx#known-limitations)

>While MDX2 supports a variety of runtimes ([React](https://mdxjs.com/packages/react/), [Preact](https://mdxjs.com/packages/preact/), [Vue](https://mdxjs.com/packages/vue/)), Storybook’s implementation is React-only. That means your documentation is rendered in React, while your stories render in the runtime of your choice (React, Vue, Angular, Web Components, Svelte, etc.).

so for more complex components (and their stories), I need to extract those stories from the mdx files.
 On a side note,  while technically this still works, putting children inside the canvas element is deprecated. 


While @bnjm's PR is getting us halfway there, I'll open a fresh one with similar changes plus the mdx/story refactorings just because I merged a bunch of non-storybook dependency updates to main already,  and there are many conflicts in `package-lock.json`

**Pull request closed without merging**